### ISSUE TITLE
Bedrock Server is not listening to Ports

### ISSUE THREAD

**Issue by CrashStyler** [TheRemote/MinecraftBedrockServer#70]
I am on Ubuntu Server 18.04 LTS and I used the script here to install an Bedrock Server.
I am hosting this Server on a VPS so all Ports are unblocked by any type of firewall.
I used the ufw to enable TCP / UDP on 19132, 19133 and custom ports.
After startup of the Bedrockserver " lsof -i -P -n | grep LISTEN " run as sudo and local user doesnt show the ports which should be open. 
External pings are not working either. But if I setup a java server on any port this port will correctly open and the server will listen.
![image](https://user-images.githubusercontent.com/43825125/108403806-6a6ccd00-721f-11eb-986b-ab7f856c9861.png)
![image](https://user-images.githubusercontent.com/43825125/108403920-87a19b80-721f-11eb-9a57-b11f895356f6.png)
![image](https://user-images.githubusercontent.com/43825125/108403967-95efb780-721f-11eb-8c79-acba452e4f47.png)
![image](https://user-images.githubusercontent.com/43825125/108404040-af90ff00-721f-11eb-8ae8-84e51dc727b9.png)
![image](https://user-images.githubusercontent.com/43825125/108404248-ebc45f80-721f-11eb-909a-c246cb8d126e.png)

**Comment by naturecodevoid** [comment]
This is probably not related to this project. I’m guessing it’s a BDS bug.

**Comment by NanoCode012** [comment]
Hello @CrashStyler, I got this problem too. Did you manage to solve it? 

To my understanding, the server uses UDP which cannot ping via openport tool. You need to use something like nmap. So.. it "should?" be working.. 

From `netstat`,
![image](https://user-images.githubusercontent.com/9899957/109443947-89186280-7a6e-11eb-8577-65201573ada2.png)


But in the end, I cannot connect to it via my ios device.

**Comment by CrashStyler** [comment]
Actually the Bedrock Server listens to the UDP Ports mentioned on startup. If not create rules for ufw (if you are using it) and enable these specific 2 ports for udp mentioned in the server.properties. After startup netstat etc. doesnt show the open udp ports but you can catch traffic with tcpdump. If you are searching on google for an udp port scanner (online) you can select the 19132 port and the server responds (it means it IS listening) that it received empty udp packages. Conclusion : Bedrock Server is up and running and listening but you can‘t join worlds. It is a buggy mess (alpha) and they wang us to shift over to realms. Long live java servers

Am 01.03.2021 um 03:14 schrieb NanoCode012 <notifications@github.com>:

﻿

Hello @CrashStyler<https://github.com/CrashStyler>, I got this problem too. Did you manage to solve it?

To my understanding, the server uses UDP which cannot ping via openport tool. You need to use something like nmap. So.. it "should?" be working..

From netstat,
[image]<https://user-images.githubusercontent.com/9899957/109443947-89186280-7a6e-11eb-8577-65201573ada2.png>

But in the end, I cannot connect to it via my ios device.

**Comment by NanoCode012** [comment]
I forgot to leave my reply back. After leaving it for a while and rejoining the server many times, it suddenly worked? Not sure if latency or something else... I usually have to connect a few times to get in...

**Comment by TheRemote** [comment]
Hey everyone,

Sorry for the super slow responses, I'm trying to get caught up on my GitHub projects here.  This has come up in the comments on my site though for the Bedrock server.

It is a BDS bug and one that has been going on for a while now unfortunately.  Here's some bugs related to it:
https://bugs.mojang.com/projects/BDS/issues/BDS-1094
https://bugs.mojang.com/projects/BDS/issues/BDS-10608
https://bugs.mojang.com/browse/BDS-3989

As NanoCode alluded to in his post right before mine restarting the server can help but what we've found is that the order in which you start the server matters too.  Try starting the servers in different orders.  I can't remember if you want to start the one using the default ports first (if there is one) or if you want to start it last, but we found the order of startup to have an impact!

Changing the order you start them up is the only thing that we've found that can have an impact and even then it feels pretty janky.  These bugs have been going on for a while now and are still in "open" status unfortunately!
